activation/activation.go

```diff
@@ -623,19 +617,8 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 		time.Until(b.layerClock.LayerToTime(atx.TargetEpoch().FirstLayer())),
 	)
 
-	select {
-	case <-atxReceived:
-		logger.With().Info("received atx in db", atx.ID())
-		if err := b.discardChallenge(); err != nil {
-			return fmt.Errorf("%w: after published atx", err)
-		}
-	case <-b.layerClock.AwaitLayer((atx.TargetEpoch()).FirstLayer()):
-		if err := b.discardChallenge(); err != nil {
-			return fmt.Errorf("%w: publish epoch has passed", err)
-		}
-		return fmt.Errorf("%w: publish epoch has passed", ErrATXChallengeExpired)
-	case <-ctx.Done():
-		return ctx.Err()
+	if err := b.discardChallenge(); err != nil {
+		return fmt.Errorf("%w: after published atx", err)
```

Comment:
NIT, the error message could be more helpful:
```suggestion
		return fmt.Errorf("discarding challenge after published ATX: %w", err)
```